test/jdk/jdk/nio/zipfs/ZipFSDirectoryExceptionMessageTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @test
+ * @bug 8305945
+ * @summary Validate that Zip FS provides the correct exception message
+ * @modules jdk.zipfs
+ * @run junit ZipFSDirectoryExceptionMessageTest
+ */
+public class ZipFSDirectoryExceptionMessageTest {
+    /**
+     * Name of Directory created within the Zip file
+     */
+    public static final String DIRECTORY_NAME = "folder/";
+    /**
+     * The expected error message
+     */
+    public static final String DIR_EXCEPTION_MESSAGE = "/folder: is a directory";
+    /**
+     * Zip file to create
+     */
+    public static final String ZIP_FILE = "directoryExceptionTest.zip";
```

Comment:
Maybe ZIP_FILE could be a Path, to avoid Path.of(ZIP_FILE) wrapping later in the test?

Reply:
I often do that but chose not to when I created the test but have updated per your suggestion